WordPress/WordPressTest/ViewRelated/Post/Utils/PostNoticeViewModelTests.swift

```diff
@@ -24,114 +64,166 @@ class PostNoticeViewModelTests: XCTestCase {
         super.tearDown()
     }
 
-    func testNoticesToBeShownAfterFailingToUploadPosts() {
-        struct Expectation {
-            let scenario: String
-            let post: Post
-            let isInternetReachable: Bool
-            let title: String
-            let actionTitle: String
+    // MARK: - Expectations
+
+    private func verify(_ scenario: Scenario) -> Nimble.ToSucceedResult {
```

Comment:
I had a veeery long discussion with @shiki in the past about Parameterized Testing here.

My argument at the time was that I preferred a lot of unit tests/code (aka `func`) but easy to read versus saving lines of code in the favor of abstraction — which IMHO makes it harder to read.

Now I'm more worried because we have more lines of code just to do the asserts and different entities. 😬 Someone in the future will probably be worried by the functions and entities in this test, it's not as simple as just reading a single test.

Anyway, I don't think we need to refactor the entire thing. But keep that in mind.

I prefer clear tests with more lines of code than abstracting it (which I believe shouldn't be in unit tests at all).

Reply:
Just for context, we discussed this and agreed to move ahead with the merge.

For extra context: I do agree that it feels like we have a framework just for these tests, and it would be better to simplify a bit.  That said, the main reason I introduced these changes was so that I don't have to add the page tests into the single method we had for testing.

Reply:
Oof. Yeah, the `Expectation` was an experiment, really. I had planned for the design to grow to be reusable as a parameterized test framework within the unit tests. Obviously, it wasn't reusable in `PostNoticeViewModelTests` since the other goal was to make it isolated. 😄 

But I'd agree that it probably wasn't the best way to do it. Perhaps discussions should have happened first. 🤷‍♂ 

Parameterized tests are great for testing a lot of similar things in one go. I'd agree that to a point, it does get a little bit tricky to read and understand. On the other hand, having individual methods for all of these would also make it difficult to identify that they're all related and testing just one path. 

I don't think @leandroalonso and I ever agreed on this. The agreement wasn't that important when you consider the overarching goal during the time this was created → finishing Offline Posting. At that time, any test was better than no test! 😄 

I'm all for killing this code if it comes down to it. My feelings will not be hurt. 😄